### DISCUSSION TITLE
Sample in agent and oap-server

### DISCUSSION THREAD

**Discussion by BritneyStars** [apache/skywalking#10399] [Q&A]
I have a question about Sample in agent and oap-server.  In agent, when the trace segment is not be sampled, the trace records will be discarded and it was not be reported the kafka? In oap-server, if setting the trace sampling, it only influence the segment data ,but the other oal metrics such as count,duration are actual and accurate exception the relation metrics. Above these is right?

**Comment by wu-sheng** [comment]
All metrics including topology(relation) are accurate if use backend sampling.
And agent sampling will abandon the raw data, so metrics lost.

**Reply by BritneyStars** [reply]
ok, I known ,thank you
